geomstats/geometry/connection.py

```diff
@@ -477,6 +477,62 @@ def directional_curvature(
         return self.curvature(tangent_vec_a, tangent_vec_b, tangent_vec_a,
                               base_point)
 
+    def curvature_derivative(
+            self, tangent_vec_u, tangent_vec_x, tangent_vec_y, tangent_vec_z,
+            base_point=None):
+        r"""Compute the covariant derivative of the curvature.
+
+        For four tangent vectors at a base point :math: `V, X, Y, Z`,
+        the covariant derivative of the curvature :math: `(\nabla_U R)(X, Y)Z`
+        is computed at the base point using Leibniz formula.
+
+        Parameters
+        ----------
+        tangent_vec_u : array-like, shape=[..., n, n]
```

Comment:
But the points can actually be vectors of size dim, square matrices for Lie groups or non-square matrices as  in Kendall shape space. Shouldn't we use dim as a proxy for (k_landmarks, m_ambient) or (n x n) when the point has matrix type?

Reply:
Yes, that's a good point: we have many cases to cover, and complicated notations might not help the user. 

Yet, so far we have made the distinction between `[..., dim]` and `[..., n, m]` to emphasize that the `tangent_vec` can have an `ndim` of 1, 2, or 3 depending on the cases.

Maybe: `shape=[..., {dim, [n, m]}]` could be a good compromise here?

Reply:
As for the [discussion on the fiber bundle class](https://github.com/geomstats/geomstats/pull/1014#discussion_r646328563), I believe that we need to refer to something like `object_shape` without caring at this level if the space is encoded as a vector or a matrix. Planning to have a class parameter `shape` to which we could refer would solve the issue but this will be for another PR.